docs/Internals.md

```diff
@@ -1,150 +1,165 @@
+            If SK EKR DOES NOT exist in metadata persistence
+                THROW ERROR: Unable to decrypt IK, missing SK from metadata
+            Use MK in HSM to decrypt SK
+            If allowed by policy, add SK to protected memory cache
+        If SK is expired
+            # NOTE: None of these currently implemented
+            Send notification SK is expired
+            Queue SK for rotation
+            Queue IK for rotation
+            Queue DRK for rotation
+        Use SK to decrypt IK
+        If allowed by policy, add IK to protected memory cache
+    If IK is expired
+        # NOTE: None of these currently implemented
+        Send notification IK is expired
+        Queue IK for rotation
+        Queue DRK for rotation #We'll continue to wind up here until we write with valid key
+    Use IK to decrypt DRK
+    Use DRK to decrypt Data
+    If DRK is expired
+        # NOTE: Not currently implemented
+        Queue DRK for rotation
+
+
+### Future Consideration: Queued Rotation
+
+Below are the proposed queue rotation flows.
+
+#### MK Rotation
+
+    This happens annually
+    Update the policy to expire all the keys
+    Once it does:
+        Queue All SKs for rotation
+        Queue All IKs for rotation
+        Queue All DRKs for rotation #Specific for each user - this is stored in the application
+
+
+#### SK Rotation
+
+    Read message from FIFO SK_IK key rotation queue
+    If SK message meta = current SK meta in metadata persistence 
+        Load SK EKR from metadata persistence 
+        Use MK in HSM to create and encrypt a new SK
+        Create and write new SK EKR in metadata persistence 
+    Delete message
+
+#### IK Rotation
+
+    Read message from FIFO SK_IK key rotation queue
+    If IK meta in message = current IK in metadata persistence
+        If SK EKR DOES NOT exist in metadata persistence 
+            THROW ERROR: no SK exists
+        Load current SK EKR from metadata persistence
+        Use MK in HSM to decrypt SK
+        If SK is expired
+            Queue SK for rotation
+            Queue IK for rotation
+        Else 
+            Create new IK from crypto library (i.e. openssl)
+            Use SK to encrypt IK
+            Create and write new IK EKR in metadata persistence 
+    Delete message
+
+#### DRK Rotation - POTENTIAL RACE CONDITION
+
+    Read message from standard DRK key rotation queue
+    Load DRK EKR from message
+    If IK is not cached 
+        Load current IK from metadata persistence 
+        If SK in IK EKR is not cached
+            Load current SK from metadata persistence 
+            Use MK in HSM to decrypt SK
+        If SK is expired
+            Queue SK for rotation
+            Queue IK for rotation
+            Exit #We'll be back once SK has rotated
+        Use SK to decrypt IK
+    If IK is expired
+        Queue IK for rotation
+        Exit  #We'll be back once IK has rotated
+    Create new DRK from crypto library (i.e. openssl)
```

Comment:
nit: replace `i.e.` with `e.g.` (think we missed in the original changes prior to moving to public repo)